tasks/install_powershell.ps1

```diff
@@ -0,0 +1,50 @@
+[CmdletBinding()]
+Param(
+	[String]$version
+)
+# If an error is encountered, the script will stop instead of the default of "Continue"
+$ErrorActionPreference = "Stop"
+
+$arch = if ((Get-WmiObject Win32_OperatingSystem).OSArchitecture -match '^32') {"x86"} else {"x64"}
+
+if ($version) {
+    $msi_source = "http://downloads.puppetlabs.com/windows/puppet5/puppet-agent-${version}-${arch}.msi"
+}
+else {
+    $msi_source = "http://downloads.puppetlabs.com/windows/puppet5/puppet-agent-${arch}-latest.msi"
+}
+
+$msi_dest = Join-Path ([System.IO.Path]::GetTempPath()) "puppet-agent-$arch.msi"
+
+function DownloadPuppet {
+  Write-Verbose "Downloading the Puppet Agent for Puppet Enterprise on $env:COMPUTERNAME..."
+
+  $webclient = New-Object system.net.webclient
+
+  try {
+    $webclient.DownloadFile($msi_source,$msi_dest)
+  }
+  catch [System.Net.WebException] {
+    # If we can't find the msi, then we may not be configured correctly
+    if($_.Exception.Response.StatusCode -eq [system.net.httpstatuscode]::NotFound) {
+        Throw "Failed to download the Puppet Agent installer: $msi_source"
+    }
+    # Throw all other WebExceptions
+    Throw $_
+  }
+}
+
+function InstallPuppet {
+  $msiexec_args = "/qn /i $msi_dest"
+  Write-Output "Installing the Puppet Agent on $env:COMPUTERNAME..."
+  $msiexec_proc = [System.Diagnostics.Process]::Start('msiexec', $msiexec_args)
+  $msiexec_proc.WaitForExit()
+  if (@(0, 1641, 3010) -NotContains $msiexec_proc.ExitCode) {
+    Throw "Something went wrong with the installation on $env:COMPUTERNAME. Exit code: " + $msiexec_proc.ExitCode
+  }
+}
+
+DownloadPuppet
+InstallPuppet
```

Comment:
Should this script cleanup the downloaded file afterwards?

Reply:
Added a `Cleanup` function to remove MSI and Install Log upon successful installation.